dspace-api/src/main/java/org/dspace/profile/ResearcherProfileServiceImpl.java

```diff
@@ -143,13 +151,11 @@ public Item findByIdentifier(Context context, Map<String, String> identifiers)
                 .collect(Collectors.joining(" OR ")
             )
         );
-
         DiscoverResult discoverResult = search(context, discoverQuery);
         List<IndexableObject> indexableObjects = discoverResult.getIndexableObjects();
-        if (CollectionUtils.isEmpty(indexableObjects)) {
-            return null;
-        }
-        return (Item) indexableObjects.get(0).getIndexedObject();
+        return indexableObjects.stream()
+            .map(object -> new ResearcherProfile((Item) object.getIndexedObject(), false))
+            .collect(Collectors.toList());
```

Comment:
Don't know why you don't like `.toList()` shortcut. 
* `toList()` return an immuable list
* `collect(Collectors.toList())` return a mutable list. But in mossssssst of the case, we will never update this kind of returned list